libvirt/libvirt_test.go

```diff
@@ -96,12 +107,82 @@ func TestLibirt(t *testing.T) {
 		expectedType := make(map[string]int64)
 		So(cpu, ShouldHaveSameTypeAs, expectedType)
 	})
+	Convey("Get Network Statistics without network interface neither path", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		domains, err := GetInstances(conn)
+		network, err := GetNetworkStatistics(domains[0])
+		So(err, ShouldNotBeNil)
+		emptyMap := make(map[string]libvirt.VirDomainInterfaceStats)
+		So(network, ShouldResemble, emptyMap)
+	})
+	Convey("Get Network Statistics with one path", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		So(err, ShouldBeNil)
+		_, err = conn.DomainCreateXMLFromFile("./test_domain_multiple_interfaces.xml", 0)
+		So(err, ShouldBeNil)
+		defer conn.CloseConnection()
+		name := "instance-00000031"
+		domain, err := GetInstanceByDomainName(conn, name)
+		So(err, ShouldBeNil)
+		network, err := GetNetworkStatistics(domain, "tap88709cbd-90")
+		So(network["tap88709cbd-90"].RxBytes, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].RxPackets, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].RxDrop, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].RxErrs, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].TxBytes, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].TxDrop, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].TxErrs, ShouldNotBeNil)
+		So(network["tap88709cbd-90"].TxPackets, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+		emptyMap := make(map[string]libvirt.VirDomainInterfaceStats)
+		So(network, ShouldHaveSameTypeAs, emptyMap)
+		So(network, ShouldNotBeEmpty)
+	})
+	Convey("Get Network Statistics with error", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		defer conn.CloseConnection()
+		conn.DomainCreateXMLFromFile("./test_domain_multiple_interfaces.xml", 0)
+		So(err, ShouldBeNil)
+		domains, err := GetInstances(conn)
+		block, err := GetNetworkStatistics(domains[0], "wrong")
+		emptyMap := make(map[string]libvirt.VirDomainInterfaceStats)
+		So(block, ShouldResemble, emptyMap)
+		So(err, ShouldNotBeNil)
+	})
+	Convey("Get Block Statistics", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		conn.DomainCreateXMLFromFile("test_domain.xml", 0)
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		name := "instance-00000031"
+		domain, err := GetInstanceByDomainName(conn, name)
+		block, err := GetBlockStatistics(domain)
+		So(err, ShouldBeNil)
+		emptyMap := make(map[string]libvirt.VirDomainBlockStats)
+		So(block, ShouldHaveSameTypeAs, emptyMap)
```

Comment:
The type of tests makes it impossible to check specific values returned by function